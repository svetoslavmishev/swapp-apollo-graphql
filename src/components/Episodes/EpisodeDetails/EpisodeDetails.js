import React, { useContext } from 'react';
import { useParams } from 'react-router-dom';
import { useQuery } from '@apollo/react-hooks';
import { useHistory } from 'react-router-dom';
import {
  Button,
  Card,
  CardContent,
  Typography,
  CardMedia
} from '@material-ui/core';

import { Header, CharCard } from '../../index';
import Loading from '../../shared/Loading/Loading';
import { GET_EPISODE } from '../../../queries/queries';
import { ThemeContext } from '../../../themeContext';
import styles from './EpisodeDetailsStyles';

const EpisodeDetails = () => {
  const { episodeId } = useParams();
  const { currentTheme } = useContext(ThemeContext);
  const classes = styles({ currentTheme });
  const history = useHistory();
  const { data, loading, error } = useQuery(GET_EPISODE, {
    variables: { id: episodeId }
  });

  if (loading) return <Loading />;
  if (error) return null;

  return (
    <>
      <Header />
      <div className={classes.cardContainer}>
        <Card className={classes.card}>
          <CardMedia
            className={classes.cover}
            image={data.episode.image}
            title={data.episode.title}
          />
          <div className={classes.details}>
            <CardContent className={classes.content}>
              <Typography classes={{ root: classes.episodeTitle }} variant="h5">
                {data.episode.title}
              </Typography>
            </CardContent>
          </div>
        </Card>
        <Card className={classes.card}>
          <div className={classes.details}>
            <CardContent>
              <Typography classes={{ root: classes.description }}>
                {data.episode.openingCrawl}
              </Typography>
              <Typography classes={{ root: classes.description }}>
                Director:{' '}
                <span className={classes.descriptionResult}>
                  {data.episode.director}
                </span>
              </Typography>
              <Typography classes={{ root: classes.description }}>
                Release date:
                <span className={classes.descriptionResult}>
                  {data.episode.releaseDate}
                </span>
              </Typography>
            </CardContent>
          </div>
        </Card>
        <CardContent className={classes.characters}>
          {data &&
            data.episode.people.edges.map(char => {
              return (
                <CharCard
                  key={char.node.id}
                  chars={char.node}
                  onClick={() => history.push(`/characters/${char.node.id}`)}
                />
              );
            })}
        </CardContent>
        <div className="flex justify-center">
          <Button
            className={classes.button}
            variant="contained"
            //onClick={loadMore}
          >
            Load more
          </Button>
        </div>
      </div>
    </>
  );
};

export default EpisodeDetails;
