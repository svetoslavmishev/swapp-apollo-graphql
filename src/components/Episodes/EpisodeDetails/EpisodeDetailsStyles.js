import { makeStyles } from '@material-ui/styles';

const styles = makeStyles(theme => ({
  cardContainer: {
    padding: '30px 100px',
    display: 'flex',
    justifyContent: 'center',
    flexDirection: 'column',
    background: ({ currentTheme }) =>
      theme.palette[currentTheme].backgroundColor
  },
  card: {
    display: 'flex',
    margin: 4,
    background: ({ currentTheme }) =>
      theme.palette[currentTheme].Cards.backgroundColor
  },
  characters: {
    display: 'flex',
    justifyContent: 'space-around',
    flexWrap: 'wrap'
  },
  content: {
    width: '100%',
    border: ({ currentTheme }) => theme.palette[currentTheme].Cards.borderColor
  },
  cover: {
    minWidth: 200,
    minHeight: 200,
    color: ({ currentTheme }) =>
      theme.palette[currentTheme].PrimaryHeading.fontColor
  },
  episodeTitle: {
    fontFamily: 'StarWars',
    color: ({ currentTheme }) =>
      theme.palette[currentTheme].PrimaryHeading.fontColor
  },
  description: {
    color: ({ currentTheme }) => theme.palette[currentTheme].textColor
  },
  descriptionDetails: {
    '& span:nth-child(even)': {
      color: theme.palette.commonColors.lightBlue
    },
    '& span:nth-child(odd)': {
      color: ({ currentTheme }) =>
        theme.palette[currentTheme].PrimaryHeading.fontColor
    }
  },
  button: {
    fontFamily: 'StarWars',
    background: ({ currentTheme }) =>
      theme.palette[currentTheme].SolidButtons.backgroundColor,
    color: ({ currentTheme }) =>
      theme.palette[currentTheme].SolidButtons.fontColor,
    margin: 40,
    '&:hover': {
      color: '#000000',
      background: theme.palette.commonColors.yellow
    }
  }
}));

export default styles;
