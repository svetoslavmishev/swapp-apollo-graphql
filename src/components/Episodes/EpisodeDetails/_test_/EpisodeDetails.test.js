import React from 'react';
import { shallow } from 'enzyme';
import { ThemeContext, theme } from '../../../../themeContext';
import EpisodeDetails from '../EpisodeDetails';

describe('<EpisodeDetails />', () => {
  it('renders <EpisodeDetails /> without error', () => {
    const wrapper = shallow(
      <ThemeContext.Provider value={theme}>
        <EpisodeDetails />
      </ThemeContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
