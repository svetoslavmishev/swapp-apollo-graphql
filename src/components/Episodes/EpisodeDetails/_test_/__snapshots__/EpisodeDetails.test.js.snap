// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EpisodeDetails /> renders <EpisodeDetails /> without error 1`] = `ShallowWrapper {}`;
